services/job_chat/job_chat.py

```diff
@@ -1,96 +1,145 @@
 import os
-from anthropic import Anthropic
-from util import DictObj, create_logger
+from typing import List, Optional, Dict, Any
+from dataclasses import dataclass
+from anthropic import (
+    Anthropic,
+    APIConnectionError,
+    BadRequestError,
+    AuthenticationError,
+    PermissionDeniedError,
+    NotFoundError,
+    UnprocessableEntityError,
+    RateLimitError,
+    InternalServerError,
+)
+from util import ApolloError, create_logger
 from .prompt import build_prompt
 
 logger = create_logger("job_chat")
 
-ANTHROPIC_API_KEY = os.getenv(
-    "ANTHROPIC_API_KEY",
-)
-
-# claude_model = "claude-3-haiku-20240307"
-claude_model = "claude-3-5-sonnet-20240620"
-max_tokens = 1024
 
+@dataclass
+class Payload:
+    """
+    Data class for validating and storing input parameters.
+    Required fields will raise TypeError if not provided.
+    """
 
-class Payload(DictObj):
-    api_key: str
     content: str
-    # history # list of {role , content } dicts
-    # context { expression, adaptor, input, output, log  }
-
+    context: Optional[str] = None
+    api_key: Optional[str] = None
 
-def main(dataDict) -> dict:
-    data = Payload(dataDict)
-    result = generate(
-        data.content, dataDict["history"] if "history" in dataDict else [], data.context, data.get("api_key")
-    )
-    return result
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "Payload":
+        """
+        Create a Payload instance from a dictionary, validating required fields.
+        """
+        if "content" not in data:
+            raise ValueError("'content' is required")
 
+        return cls(content=data["content"], context=data.get("context"), api_key=data.get("api_key"))
 
-def generate(content, history, context, api_key) -> str:
-    if api_key is None and isinstance(ANTHROPIC_API_KEY, str):
-        logger.warn("Using default API key from environment")
-        api_key = ANTHROPIC_API_KEY
 
-    client = Anthropic(api_key=api_key)
+@dataclass
+class ChatConfig:
+    model: str = "claude-3-5-sonnet-20240620"
+    max_tokens: int = 1024
+    api_key: Optional[str] = None
 
-    logger.info("Anthropic client loaded")
-    (system_message, prompt) = build_prompt(content, history, context)
-
-    # Useful in local dev but contains a lot of stuff, so uncommenting
-    # logger.info("")
-    # logger.info("--- SYSTEM ---")
-    # logger.info(system_message)
-    # logger.info("--------------")
-    # logger.info("")
-    # logger.info("--- PROMPT ---")
-    # logger.info(prompt)
-    # logger.info("--------------")
-    # logger.info("")
-
-    try:
-        logger.info("Generating")
 
-        message = client.beta.prompt_caching.messages.create(
-            max_tokens=max_tokens, messages=prompt, model=claude_model, system=system_message
+@dataclass
+class ChatResponse:
+    content: str
+    history: List[Dict[str, str]]
+    usage: Dict[str, Any]
+
+
+class ClaudeClient:
```

Comment:
`AnthropicClient` would be a better class name. I'm learning (the hard way) that Claude and the Anthropic API are two very different products!